### DISCUSSION TITLE
help：Selecting the best reference date problem

### DISCUSSION THREAD

**Discussion by Rick0219** [alexisInSAR/EZ-InSAR#36] [Help]
Hi,Alexis
I have met some problems when I use the button "Select the best reference date problem".I choose POD and potinel reference date.

![image](https://user-images.githubusercontent.com/93982725/234905685-a7a5e110-0d37-4f1a-9af1-72ef8d80ce61.png)

**Comment by alexisInSAR** [comment]
Hi, 

Does it work if you use the default settings?

Kind regards, 

Alexis

**Reply by Rick0219** [reply]
Thanks for your reply.But it shows the same problem while using default settings.And I know I can just skip this step and choose a random date(like what I did).But I do want to know what makes this problem?

**Reply by anandsar01** [reply]
Hi @alexisInSAR 
I got the same error. It neither working on defaults settings nor for POD.

**Comment by Sandeepkudipudi** [comment]
HI sir,
I am also faced the same error at selecting the best reference date.
I had even tried using the default parameters and also by giving some reference date and changing to POD.
Please help me with this.

**Comment by Rick0219** [comment]
I heard that ESA has moved all its service to dataspace.copernicus.eu,maybe this is why we can not download the  POD now?

**Reply by Rick0219** [reply]
@alexisInSAR

**Comment by Sandeepkudipudi** [comment]
Thank you for the response,
So, now if I download the POD files externally and kept them in the orbit folder of the project folder, then i'll get the coarse network of interferograms at the stage of selection of the best reference date.

**Comment by alexisInSAR** [comment]
Hi all, 

The new version of EZ-InSAR allows to download the orbit files from ASF (https://github.com/alexisInSAR/EZ-InSAR/discussions/72). 

Let me know if it works. 

Kind regards, 

Alexis

**Reply by anandsar01** [reply]
Hi @alexisInSAR 

I manually downloaded the orbit files for the all the SLCs and put them into the subfolders (named them as their respective date) in the orbit folder.
Still I am unable to **select best reference date** and getting this error. I tried running this command on both- default value and giving a POD
![01](https://github.com/alexisInSAR/EZ-InSAR/assets/154870709/523b35b9-efd2-4da2-bacb-b5918bcfc9c9)

**Comment by euyurioliveira** [comment]
Hello everyone,

I'm having an issue trying to perform the "Select the best reference date," as presented below.

![Captura de tela de 2024-02-25 08-49-53](https://github.com/alexisInSAR/EZ-InSAR/assets/108848153/b4d0427e-d60d-4836-b7a5-1f473fafe197)

I kept the default options for Orbit MODE and Potential Reference Date ("None"). Additionally, upon checking the target.kml file, there doesn't seem to be any problem, as shown below.

![Captura de tela de 2024-02-25 08-59-36](https://github.com/alexisInSAR/EZ-InSAR/assets/108848153/da83a295-b910-48ca-a48d-5e9d5ce02db3)

Does anyone have any suggestions for solving this error? 
I appreciate everyone's help in advance.

**Reply by alexisInSAR** [reply]
Hi @yurioliveira-ctec, 

Your error is related to your Python installation. Please check your installation of the Fiona package. 

Kind regards, 

Alexis

**Reply by euyurioliveira** [reply]
Hi Alexis,

I am grateful for the response and attention. I managed to solve the problem I mentioned, apparently in my case the 'kml' driver was not activated by default. So I had to add these lines of code, shown below, in the coarse_Sentinel_1_baselines.py file.

fiona.drvsupport.supported_drivers['kml'] = 'rw' 
fiona.drvsupport.supported_drivers['KML'] = 'rw'
fiona.drvsupport.supported_drivers['libkml'] = 'rw'
fiona.drvsupport.supported_drivers['LIBKML'] = 'rw'

If anyone knows a way to activate the drivers globally, I would be grateful.

Eagerly waiting for the new version of EZ-InSAR as it is facing problems with Alaska servers (https://github.com/alexisInSAR/EZ-InSAR/discussions/80).

I thank the attention!

**Reply by forgetsyl** [reply]
Hi, can you elaborate on where to add the code. I tried without success. Thank you!

**Comment by emhau** [comment]
Hi Alexis

I also have an issue with selecting the best reference date. I'm using the newest version (2.2.2) of EZ-InSAR and downloaded the orbit files from the Copernicus server. I've tried using both "POD" and "none" as inputs in ORBIT mode and "none" as a potential reference date. Here's the print of my error message. 
   
![image](https://github.com/alexisInSAR/EZ-InSAR/assets/163998236/2b5d49cc-4e85-45f4-afb5-8898ad5d681f)


Thank you in advance, 
Emma

**Reply by alexisInSAR** [reply]
Hi Emma, 

I think there is an error with an orbit file. Can you show me the first lines of your MATLAB terminal when you use EZ-InSAR? 

Alexis

**Reply by emhau** [reply]
Hi! Heres the lines when I open Ez-InSAR
![image](https://github.com/alexisInSAR/EZ-InSAR/assets/163998236/46b45c9d-b5df-4c9a-be80-d3bc00843dcc)

**Comment by forgetsyl** [comment]
After a lot of failed attempts, I finally got it to work. 
The issue was with the source of the KML file. 
I hope that what I've learned will help you all.

**Reply by Agupta662** [reply]
Hi @forgetsyl! Can you elaborate on how you fixed the source of KML file?

**Reply by forgetsyl** [reply]
First, you need to look at all the errors that were recorded and try to fix them. This is because the errors could be caused by other issues, such as missing cv2 or fiona packages. I fixed all the problems but still got errors. I then looked at all my kml files and thought something was off. I tried to make a 15km x 10km kml file again using Google Earth (and a QGIS shapefile), and it worked.
